fee_calculator/apps/api/tests/test_scheme_api.py

```diff
@@ -33,7 +38,31 @@ def test_get_by_date_available(self):
         response = self.client.get('%s?type=AGFS&case_date=2012-04-02' % self.endpoint)
         self.assertEqual(response.status_code, status.HTTP_200_OK)
         self.assertEqual(len(response.data['results']), 1)
-        self.assertEqual(response.data['results'][0]['id'], 1)
+        self.assertEqual(response.data['results'][0]['id'], AGFS_SCHEME_NINE_ID)
+
+    def test_get_by_date_available_agfs_12_no_hearing_date(self):
+        response = self.client.get('%s?type=AGFS&case_date=2020-09-17' % self.endpoint)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(response.data['results']), 1)
+        self.assertEqual(response.data['results'][0]['id'], AGFS_SCHEME_TWELVE_ID)
+
+    def test_get_by_date_available_agfs_12_with_main_hearing_date(self):
+        response = self.client.get('%s?type=AGFS&case_date=2020-09-17&main_hearing_date=2022-10-30' % self.endpoint)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(response.data['results']), 1)
+        self.assertEqual(response.data['results'][0]['id'], AGFS_SCHEME_TWELVE_ID)
+
+    def test_get_by_date_available_agfs_12a(self):
```

Comment:
Should this be `agfs_clair_contingency` rather than `agfs_12a`? (Looks like this might be a hangover from the spike where we used the 12a naming convention)